src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,526 @@
+        const alertMessage = this.state.alertMessage || CalendarContainer.validateProps(this.props);
+
+        return createElement("div",
+            {
+                style: this.state.loading ? { ...parseStyle(this.props.style) } : undefined
+            },
+            createElement(Calendar, {
+                alertMessage,
+                messages: this.setCustomViews(),
+                events: this.state.events,
+                defaultView: this.props.defaultView,
+                dayFormat: this.props.dayFormat,
+                weekdayFormat: this.props.weekdayFormat,
+                timeGutterFormat: this.props.timeGutterFormat,
+                monthHeaderFormat: this.props.monthHeaderFormat,
+                dayHeaderFormat: this.props.dayHeaderFormat,
+                loading: this.state.loading,
+                popup: this.props.popup,
+                resizable: this.props.resizable,
+                startPosition: this.state.startPosition,
+                selectable: !readOnly ? this.props.selectable : false,
+                views: this.props.views,
+                onSelectEventAction: !readOnly ? this.onClickEvent : undefined,
+                onEventResizeAction: !readOnly ? this.onResizeEvent : undefined,
+                onSelectSlotAction: !readOnly ? this.onClickSlot : undefined,
+                onEventDropAction: !readOnly ? this.onDropEvent : undefined
+            })
+        );
+    }
+
+    componentDidMount() {
+        if (!this.state.alertMessage && this.props.mxObject) {
+            this.fetchData(this.props.mxObject);
+            this.setStartPosition(this.props.mxObject);
+        }
+    }
+
+    componentWillUnMount() {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+    }
+
+    componentWillReceiveProps(nextProps: CalendarContainerProps) {
+        if (!this.state.loading) { this.setState({ loading: true }); }
+        if (!this.state.alertMessage) {
+            this.fetchData(nextProps.mxObject);
+            this.setStartPosition(nextProps.mxObject);
+            this.setRefreshInterval(nextProps.refreshInterval, nextProps.mxObject);
+        }
+        this.fetchData(nextProps.mxObject);
+        this.resetSubscriptions(nextProps.mxObject);
+    }
+
+    private setRefreshInterval(refreshInterval: number, mxObject?: mendix.lib.MxObject) {
+        if (refreshInterval > 0 && mxObject) {
+            this.clearRefreshInterval();
+            this.intervalID = window.setInterval(() => {
+                if (!this.state.loading) {
+                    this.fetchData(mxObject);
+                }
+            }, refreshInterval);
+        }
+    }
+
+    private clearRefreshInterval() {
+        if (this.intervalID) { window.clearInterval(this.intervalID); }
+    }
+
+    private setStartPosition = (mxObject: mendix.lib.MxObject) => {
+        if (this.props.startPositionAttribute !== "" && mxObject) {
+        this.setState({
+            loading: false,
```

Comment:
set state loading to false... though it not guaranteed that all calendar object are loaded